### ISSUE TITLE
External service cluster generation is broken in V1

### ISSUE THREAD

**Issue by ijsnellf** [istio/istio#3901]
Currently V1 generates clusters based on the type of the host of the rule, not the host it is being redirected to. Given:
```yaml
hosts:
  - internal.service
http:
  - route:
    - destination:
        name: internal.service
      weight: 75
    - destination:
        name: external.service.com
      weight: 25
```
where `external.service.com` is an external service,
```yaml
hosts:
  - "external.service.com"
ports:
  - number: 80
    name: http-number
    protocol: http
discovery: NONE
```
the cluster for `external.service.com` will be constructed as if the destination was an internal service (I.E., not an Original DST cluster).

This breaks any realistic configuration scenario, like https://github.com/istio/istio/issues/3878.

**Comment by linsun** [comment]
@ijsnellf do you think this would be fixed by moving to envoy v2?

**Comment by sakshigoel12** [comment]
@ijsnellf please could you confirm is this can be closed since 0.8 release is out.

**Comment by ijsnellf** [comment]
This is no longer the case in v2 with the v1alpha3 networking rules.
